File: Java_Codes/5. Arrays/initialization.java
public class initialization{

    public static void main(String[] args) {
        // Call either or both of these based on what you want to test
        demonstrate1DArrayInitialization();
        demonstrate2DArrayInitialization();
    }

    // Function to demonstrate all 1D array initialization methods
    public static void demonstrate1DArrayInitialization() {
        System.out.println("------ 1D Array Initialization ------");

        // 1. Static Initialization
        int[] staticArr = {1, 2, 3, 4, 5};
        System.out.print("Static Init: ");
        print1DArray(staticArr);

        // 2. Dynamic Initialization
        int[] dynamicArr = new int[5];
        System.out.print("Dynamic Init: ");
        print1DArray(dynamicArr);

        // 3. Declared then initialized
        int[] declaredArr;
        declaredArr = new int[]{10, 20, 30};
        System.out.print("Declared then Init: ");
        print1DArray(declaredArr);

        // 4. Initialization using loop
        int[] loopArr = new int[5];
        for (int i = 0; i < loopArr.length; i++) {
            loopArr[i] = i * 2;
        }
        System.out.print("Loop Init: ");
        print1DArray(loopArr);

        // 5. Using Java 8 Streams
        int[] streamArr = java.util.stream.IntStream.range(0, 5).toArray();
        System.out.print("Stream Init: ");
        print1DArray(streamArr);

        System.out.println(); // spacing
    }

    // Function to demonstrate all 2D array initialization methods
    public static void demonstrate2DArrayInitialization() {
        System.out.println("------ 2D Array Initialization ------");

        // 1. Static Initialization
        int[][] staticMatrix = {
            {1, 2, 3},
            {4, 5, 6}
        };
        System.out.println("Static Init:");
        print2DArray(staticMatrix);

        // 2. Dynamic Initialization
        int[][] dynamicMatrix = new int[2][3];
        System.out.println("Dynamic Init:");
        print2DArray(dynamicMatrix);

        // 3. Jagged Array
        int[][] jaggedMatrix = new int[2][];
        jaggedMatrix[0] = new int[3];
        jaggedMatrix[1] = new int[2];
        System.out.println("Jagged Init:");
        print2DArray(jaggedMatrix);

        // 4. Using nested loops
        int[][] loopMatrix = new int[2][3];
        for (int i = 0; i < loopMatrix.length; i++) {
            for (int j = 0; j < loopMatrix[i].length; j++) {
                loopMatrix[i][j] = i + j;
            }
        }
        System.out.println("Loop Init:");
        print2DArray(loopMatrix);

        // 5. Using Arrays.fill for each row
        int[][] fillMatrix = new int[2][3];
        for (int[] row : fillMatrix) {
            java.util.Arrays.fill(row, 1);
        }
        System.out.println("Arrays.fill Init:");
        print2DArray(fillMatrix);
    }

    // Helper method to print 1D arrays
    public static void print1DArray(int[] arr) {
        for (int val : arr) {
            System.out.print(val + " ");
        }
        System.out.println();
    }

    // Helper method to print 2D arrays
    public static void print2DArray(int[][] matrix) {
        for (int[] row : matrix) {
            for (int val : row) {
                System.out.print(val + " ");
            }
            System.out.println();
        }
        System.out.println(); // spacing
    }
}

File: Projects/PythonProject/Arrays/initialization.py
import numpy as np

def demonstrate_1d_array_initialization():
    print("------ 1D Array Initialization ------")

    # 1. Static Initialization
    static_arr = [1, 2, 3, 4, 5]
    print("Static Init:", static_arr)

    # 2. Dynamic Initialization (default zeros)
    dynamic_arr = [0] * 5
    print("Dynamic Init:", dynamic_arr)

    # 3. Declared then initialized
    declared_arr = None
    declared_arr = [10, 20, 30]
    print("Declared then Init:", declared_arr)

    # 4. Initialization using loop
    loop_arr = [i * 2 for i in range(5)]
    print("Loop Init:", loop_arr)

    # 5. Using range (similar to Java Streams)
    stream_arr = list(range(5))
    print("Stream Init:", stream_arr)

    print()

def demonstrate_2d_array_initialization():
    print("------ 2D Array Initialization ------")

    # 1. Static Initialization
    static_matrix = [
        [1, 2, 3],
        [4, 5, 6]
    ]
    print("Static Init:")
    print_2d_array(static_matrix)

    # 2. Dynamic Initialization
    dynamic_matrix = [[0 for _ in range(3)] for _ in range(2)]
    print("Dynamic Init:")
    print_2d_array(dynamic_matrix)

    # 3. Jagged Array
    jagged_matrix = [None] * 2
    jagged_matrix[0] = [0] * 3
    jagged_matrix[1] = [0] * 2
    print("Jagged Init:")
    print_2d_array(jagged_matrix)

    # 4. Using nested loops
    loop_matrix = [[i + j for j in range(3)] for i in range(2)]
    print("Loop Init:")
    print_2d_array(loop_matrix)

    # 5. Using fill (NumPy for simplicity)
    fill_matrix = np.full((2, 3), 1).tolist()
    print("Arrays.fill Init:")
    print_2d_array(fill_matrix)

def print_2d_array(matrix):
    for row in matrix:
        print(" ".join(map(str, row)))
    print()

def main():
    demonstrate_1d_array_initialization()
    demonstrate_2d_array_initialization()

if __name__ == "__main__":
    main()
